Include/cpython/object.h

```diff
@@ -233,6 +238,9 @@ struct _typeobject {
     /* bitset of which type-watchers care about this type */
     unsigned char tp_watched;
 
+    /* callback that may specialize BINARY_OP */
+    binop_specialize_func tp_binop_specialize;
```

Comment:
Is there a way to do this without adding a `tp_` slot?

Adding a "type slot" (`Py_tp_binop_specialize`) is fine, as it doesn't restrict how we implement it.

Reply:
I can put it in `PySequenceMethods` and `PyMappingMethods`, but then it appears twice. Is there another way?

Reply:
That doesn't help. I would prefer if it wasn't in a struct in a public header, but I don't see any other obvious place to put the data.
Maybe add a comment to the effect that this isn't part of the public API.